test/block/blockdownload.test.js

```diff
@@ -16,15 +16,850 @@
+                        'content-length': 1911,
+                        'content-disposition': 'attachment; filename="image-file-1.png"',
+                        'last-modified': 'Wed, 21 Oct 2015 07:28:00 GMT'
+                    }
+                ];
+            });
+
+        nock(HOST)
+            .get(filenameToDownload2)
+            .replyWithFile(200,
+                `${__dirname}/test-files/png-file.png`,
+                {
+                    'content-type': 'image/png',
+                    'content-length': 1911,
+                    'content-disposition': 'attachment; filename="image-file-1.png"',
+                    'last-modified': 'Wed, 21 Oct 2015 07:28:00 GMT'
+                }
+            );
+
+        const blockDownload = new BlockDownload();
+        const events = {
+            filestart: [],
+            fileprogress: [],
+            fileend: [],
+            error: []
+        };
+        blockDownload.on('filestart', (data) => {
+            events.filestart.push(data);
+        });
+        blockDownload.on('fileprogress', (data) => {
+            events.fileprogress.push(data);
+        });
+        blockDownload.on('fileend', (data) => {
+            events.fileend.push(data);
+        });
+        blockDownload.on('error', (data) => {
+            events.error.push(data);
+        });
+
+        const fileToDownload = `${HOST}${filenameToDownload}`;
+        const mockDownloadFileLocation = "./test/tmp.jpeg";
+        const fileToDownload2 = `${HOST}${filenameToDownload2}`;
+        const mockDownloadFileLocation2 = "./test/tmp.png";
+        await blockDownload.downloadFiles({
+            downloadFiles: [{
+                fileUrl: fileToDownload,
+                filePath: Path.resolve(mockDownloadFileLocation), // where to put the file
+                fileSize: 2024
+            },
+            {
+                fileUrl: fileToDownload2,
+                filePath: Path.resolve(mockDownloadFileLocation2), // where to put the file
+                fileSize: 1911
+            }],
+            concurrent: true,
+            maxConcurrent: 4
+        });
+
+        assert.equal(events.error.length, 0);
+        assert.ok(nock.isDone(), nock.pendingMocks());
+
+        const actualHash = await fileHash(mockDownloadFileLocation);
+        await fs.promises.unlink(Path.resolve(mockDownloadFileLocation), "Could not unlink mock downloaded jpeg file");
+        const expecteHash = await fileHash(`${__dirname}/test-files/jpeg-file.jpeg`);
+        assert.equal(actualHash, expecteHash);
+
+        const actualHash2 = await fileHash(mockDownloadFileLocation2);
+        await fs.promises.unlink(Path.resolve(mockDownloadFileLocation2), "Could not unlink mock downloaded png file");
+        const expecteHash2 = await fileHash(`${__dirname}/test-files/png-file.png`);
+        assert.equal(actualHash2, expecteHash2);
+    });
+
+    it('Block download two small files (different promises)', async function () {
```

Comment:
```suggestion
    it('Block download two small files (different concurrencies)', async function () {
```

Reply:
The concurrencies for the promises are the same, so changing test title would be confusing.